jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenSettingsServerCredentials.java

```diff
@@ -58,7 +59,7 @@
   @Override
   public Optional<AuthProperty> inferAuth(String registry) throws InferredAuthException {
 
-    Server server = settings.getServer(registry);
+    Server server = getServerFromMavenSettings(registry);
     if (server == null) {
```

Comment:
It'd be nice if this `server == null` code could go into getServerFromMavenSettings

Reply:
Could you elaborate? I can't see an option right now to implement that: There could, or could not be, a server returned.
If I switch this to use `Optional`, I would have to swap this out with an `isPresent` check instead to continue with the decryption below...

Reply:
Makes sense. I think we can leave this null check here.